### ISSUE TITLE
Discogs Plugin: Don't Hide Duplicate MusicBrainz (or Prefer MusicBrainz Over Discogs)

### ISSUE THREAD

**Issue by Frostflake** [beetbox/beets#3941]
Currently, it seems like a MusicBrainz release linked to a corresponding Discogs release will be hidden when using the Discogs plugin. This seems like an odd decision, since Discogs is not the default source and I prefer to use MusicBrainz for everything in my library when available.

Steps to reproduce:
1. Enable and configure the discogs plugin.
2. Import files corresponding to a release that exists on MusicBrainz with a link to the Discogs entry. (example: bbeded36-8d04-4b3b-8b8b-6fa2e4fec570)
3. After beets finds a match, press [M] to see More candidates.
4. The corresponding MusicBrainz entry will not be in the list..

Fun fact: This also happens if the MusicBrainz entry has a higher match percentage than the Discogs entry (which can be seen by typing the [I]D to select the MusicBrainz release manually)

**Comment by wisp3rwind** [comment]
Could you provide a debug log of the problematic import session and your configuration, as is asked for in the [issue template](https://github.com/beetbox/beets/blob/master/.github/ISSUE_TEMPLATE/bug-report.md)? This might give some hints at what is actually going on.

**Comment by Frostflake** [comment]
So far I have not had this happen again with my imports - it might take a while, but I'll keep an eye out for it and provide a verbose log next time it happens.

**Comment by Frostflake** [comment]
Alright, here is an example. The release I would have expected (and used for the ID lookup) is https://musicbrainz.org/release/3e94c86d-aa6d-487a-ac31-3ed0d131f816

[config.txt](https://github.com/beetbox/beets/files/6639895/config.txt)
[config-for-archive.txt](https://github.com/beetbox/beets/files/6639896/config-for-archive.txt)
[import.log](https://github.com/beetbox/beets/files/6639897/import.log)

**Comment by wisp3rwind** [comment]
So the issue here is not that the MB release would be hidden, rather, the MB query seems to return pretty bad results for this specific search. The particular ID (3e94c86d-aa6d-487a-ac31-3ed0d131f816) is actually among the results: it's entry 5 in the list shown after you selected "m"ore. It's properly sorted according to match distance. One reason that contributes to it not being listed first is the incorrect number of tracks (the MB release has 6, but beets is searching for a release with 8 tracks, since that seems to be what you're importing?). However, 3e94c86d-aa6d-487a-ac31-3ed0d131f816 is _not_ the ID that you entered when according to the log, you searched for 72dd79b7-e110-4f35-96f4-2a5726074e19, which seems to be the actual correct 8-track release. The initial MB search didn't return this release, however, so beets didn't look it up.

tl;dr: Beets is correctly displaying the releases that MB and discogs returned. However, the MB search results for this specific release don't include the expected releases. I could imagine that this is due to the various spellings of the artist's name, but someone more familiar with musicbrainz would need to verify this. I guess there's little beets can do here; ultimately this seems to be an issue with musicbrainz' search.
